sql/analyzer/memo.go

```diff
@@ -464,16 +464,21 @@ func (e *exprGroup) updateBest(n relExpr, grpCost float64) {
 	}
 }
 
-func (e *exprGroup) finalize(node sql.Node) sql.Node {
+func (e *exprGroup) finalize(node sql.Node, input sql.Schema) (sql.Node, error) {
 	props := e.relProps
 	var result = node
 	if props.filter != nil {
-		result = plan.NewFilter(props.filter, result)
+		sch := append(input, node.Schema()...)
+		filter, _, err := FixFieldIndexes(e.m.scope, nil, sch, props.filter)
```

Comment:
The fact that we need to fix up indexes suggests that maybe the rules are out of order. If we run this rule before the column indexes are applied, we could skip on this update. Is that possible?